rbac/server/api/users.py

```diff
@@ -199,13 +200,18 @@
         request:
             obj: a request object
     """
-    txn_key, txn_user_id = await utils.get_transactor_key(request)
-    is_admin = await utils.check_admin_status(txn_user_id)
-    if not is_admin:
+    try:
+        txn_key, txn_user_id = await utils.get_transactor_key(request)
+        is_admin = await utils.check_admin_status(txn_user_id)
+        if not is_admin:
+            raise ApiBadRequest(
+                "You do not have the authorization to create an account."
+            )
+        next_id = str(uuid4())
+        key_pair = Key()
+        return txn_key, txn_user_id, next_id, key_pair
+    except BadSignature:
```

Comment:
what would cause a BadSignature error to be raised in this function?

Reply:
Get transactor key raises BadSignature as it is part of the itsdangerous library.The auth decorator we have in our APIs checks for this BadSignature error, so that's why we haven't really seen the try catch for a Bad Signature within the APIs themselves. Because create_user() endpoint doesn't use the auth decorator, we catch the BadSignature error here.

Reply:
Cool, Thanks for the clarification